src/script/ismine.h

```diff
@@ -9,22 +9,44 @@
 #include <script/standard.h>
 
 #include <stdint.h>
+#include <bitset>
 
 class CKeyStore;
 class CScript;
 
 /** IsMine() return codes */
 enum isminetype
 {
-    ISMINE_NO = 0,
-    ISMINE_WATCH_ONLY = 1,
-    ISMINE_SPENDABLE = 2,
-    ISMINE_ALL = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE
+    ISMINE_NO         = 0,
+    ISMINE_WATCH_ONLY = 1 << 0,
+    ISMINE_SPENDABLE  = 1 << 1,
+    ISMINE_ALL        = ISMINE_WATCH_ONLY | ISMINE_SPENDABLE,
+    ISMINE_ENUM_ELEMENTS,
 };
 /** used for bitflags of isminetype */
 typedef uint8_t isminefilter;
 
 isminetype IsMine(const CKeyStore& keystore, const CScript& scriptPubKey);
 isminetype IsMine(const CKeyStore& keystore, const CTxDestination& dest);
 
+struct CachableAccount
```

Comment:
I just noticed the struct is called `CachableAccount` and not `CachableAmount`. This seems like an unusual choice, because while it is clear this struct is a wrapper around an amount (subdivided into spendable and watchonly parts), it's at all not clear why we'd ever think of instances of these structs as "accounts." Not even to mention in the past history of accounts in the wallet...

I'd suggest at least renaming `CachableAccount` to `CachableAmount`, though I think more ideally this would:

- Rename `CacheableAccount` to `CachedAmount`  and add doxygen comment: "Cached amount subdivided into watchonly and spendable parts."
- Rename `GetCachable` to `GetAmount`

Reply:
It is not always cached (hence the bitset), so I think "cachable" sounds better in this case. I agree on the amount vs account deal though. "An amount subdivided into spendable and watchonly parts" is exactly what this is.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/15780#discussion_r275616156 from kallewoof

> It is not always cached (hence the bitset), so I think "cachable" sounds better in this case.

Ok, but this is kind of funny logic. I wonder if you also prefer to call pointers "pointables" because sometimes they don't point to valid objects.

Reply:
If they would automatically deduct and/or instantiate the appropriate value/object when they didn't point to valid objects, they would definitely no longer just be "pointers".